src/cli/apply.py

```diff
@@ -31,15 +32,16 @@ def apply(remove):
                 asset_type)
             curr_asset_multikey_dict = hash_asset_dir_multi_hash(
                 path, HASH_ALGS)
-            install_assets(lockfile_assets_multikey_dict,
-                           lockfile_assets_multikey_dict.get_multikey_difference(
-                               curr_asset_multikey_dict),
-                           path, click.echo)
+            install_queue = create_entry_queue(
+                lockfile_assets_multikey_dict, lockfile_assets_multikey_dict.
+                get_multikey_difference(curr_asset_multikey_dict))
+
+            install_assets(install_queue, path, click.echo)
```

Comment:
the echo argument is used for errors and output, which makes them responsible for both fs modification as well as output formatting. this is ok for now, but seems messy

either:
1) make install_assets return the list of things it modifies (maybe not necessary since it should be the same as install_queue, unless there is a hash error)
2) ?